criu/clone-noasan.c

```diff
@@ -61,7 +61,14 @@ int clone3_with_pid_noasan(int (*fn)(void *), void *arg, int flags,
 
 	pr_debug("Creating process using clone3()\n");
 
-	c_args.exit_signal = exit_signal;
+	/*
+	 * clone3() explicitly blocks setting an exit_signal
+	 * if CLONE_PARENT is specified. With clone() it also
+	 * did not work, but there was no error message. The
+	 * exit signal from the thread group leader is taken.
+	 */
+	if (!(flags & CLONE_PARENT))
+		c_args.exit_signal = exit_signal;
```

Comment:
should we failt if exit_signal isn't equal to SIGCHLD?

Reply:
No idea, but if you think it would be correct, I can add that. Let me add it and push a new version of this PR.

Reply:
Yes, I think it would be correct.

Reply:
Now failing if `exit_signal != SIGCHLD`